src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/common.kt

```diff
@@ -320,56 +327,54 @@ fun refundTransaction(
   paymentGatewayClient: PaymentGatewayClient,
   refundService: RefundService,
   refundRetryService: RefundRetryService,
+  authorizationStateRetrieverService: AuthorizationStateRetrieverService,
   tracingInfo: TracingInfo?,
   retryCount: Int = 0
 ): Mono<BaseTransaction> {
+  val appendRefundedRequestEvent by lazy {
+    if (tx !is BaseTransactionWithRefundRequested) {
+      updateTransactionToRefundRequested(
+        tx, transactionsEventStoreRepository, transactionsViewRepository)
+    } else {
+      Mono.just(tx)
+    }
+  }
```

Comment:
```suggestion
  val appendRefundedRequestEvent =
    if (tx !is BaseTransactionWithRefundRequested) {
      updateTransactionToRefundRequested(
        tx, transactionsEventStoreRepository, transactionsViewRepository)
    } else {
      Mono.just(tx)
    }
```

appendRefundedRequestEvent val is a mono, no need to introduce lazy loading here